assignments/tasks_test.go

```diff
@@ -0,0 +1,539 @@
+package assignments
+
+import (
+	"context"
+	"testing"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/database"
+	"github.com/autograde/quickfeed/internal/qtest"
+	"github.com/autograde/quickfeed/scm"
+	"github.com/google/go-cmp/cmp"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+// When running tests that have anything to do with tasks/issues, it is important that issues have their title corresponding to the name of an associated task.
+// For example, if you have an issue that is supposed to be connected to the task "task-1.md" in "lab1", the title of this issue needs to be "lab1/task-1.md".
+// Otherwise when creating the database there will be no clear way to know which issue is supposed to be associated with which task.
+
+// InitializeDbEnvironment initializes a db, based on org.
+func InitializeDbEnvironment(t *testing.T, c context.Context, course *pb.Course, s scm.SCM) (database.Database, func(), error) {
+	db, cleanup := qtest.TestDB(t)
+
+	org, err := s.GetOrganization(c, &scm.GetOrgOptions{Name: course.Name})
+	if err != nil {
+		return db, cleanup, err
+	}
+
+	// Create course
+	admin := qtest.CreateFakeUser(t, db, uint64(1))
+	qtest.CreateCourse(t, db, admin, course)
+
+	// Find and create assignments
+	foundAssignments, _, err := fetchAssignments(c, zap.NewNop().Sugar(), s, course)
+	if err != nil {
+		return db, cleanup, err
+	}
+
+	err = db.UpdateAssignments(foundAssignments)
+	if err != nil {
+		return db, cleanup, err
+	}
+
+	// Creates tasks found in assignments
+	tasks := getTasksFromAssignments(c, foundAssignments)
+	for _, assignment := range foundAssignments {
+		err = synchronizeTasks(c, db, assignment, tasks)
+		if err != nil {
+			return db, cleanup, err
+		}
+	}
+
+	// Get repositories
+	repos, err := s.GetRepositories(c, org)
+	if err != nil {
+		return db, cleanup, err
+	}
+
+	// Create repositories with issues
+	n := 2
+	for _, repo := range repos {
+		var user *pb.User
+		// Might not even be necessary to handle repos differently in these tests.
+		dbRepo := &pb.Repository{}
```

Comment:
As the linter tells this allocation is never used, it is just overwritten. The correct way to do this is:
```go
		var dbRepo *pb.Repository
```